components/ai/code/ai-bbe/natural_programming_const_compile_time_code_generation_code.md

```diff
@@ -0,0 +1,21 @@
+```
+import ballerina/http;
+import ballerina/test;
+
+final http:Client shopClient = check new ("http://localhost:8080/shop");
+
+function getTestProducts() returns Product[] =>
```

Comment:
Should have been a data provider.

Reply:
We can redo the sample to something that works with data providers.

If not, let's use the test data generation in-line

e.g.,

```ballerina
    Product[] products = const natural {
        Generate lists of products with:
        - id: random id starts with "PROD", example:-  "PROD001"
        - price: random decimal between 10.0 and 1000.0
    };
    Product[] filteredProducts = filterProductsAbovePrice(products, 100);
```